src/components/organizations/show/facts.js

```diff
@@ -0,0 +1,161 @@
+import React, {Component} from 'react'
+import {connect} from 'react-redux'
+
+import Icon from 'react-fa'
+
+import Histogram from 'gComponents/simulations/histogram/index'
+
+import {simulateFact} from 'gEngine/facts'
+import {addStats} from 'gEngine/simulation'
+
+import './facts.css'
+
+
+// TODO(matthew): Also add validations for values on the server.
+
+const FactRow = ({fact}) => (
+  <div className='Fact'>
+    <div className='row'>
+      <div className='col-md-3'>
+        <Histogram
+          height={30}
+          simulation={fact.simulation}
+          cutOffRatio={0.995}
+        />
+      </div>
+      <div className='col-md-6'><span className='name'>{fact.name}</span></div>
+      <div className='col-md-2'>
+        <div className='variableName'>
+          <span className='prefix'>#</span>
+          <span className='variable'>{fact.variable_name}</span>
+        </div>
+      </div>
+      <div className='col-md-1'>
+        <span className='ui button options'><Icon name='ellipsis-v' /></span>
+      </div>
+    </div>
+  </div>
+)
+
+const hasAllNonEmpty = (obj, props) => !_.some(props.map(prop => _.isEmpty(_.get(obj, prop))))
+const readableIdPartFromWord = word => (/\d/).test(word) ? word : word[0]
+function getVariableNameFromName(rawName) {
+  const name = rawName.trim().replace(/[^\w\d]/g, ' ').toLowerCase()
+  const words = name.split(/[^\w\d]/).filter(s => !_.isEmpty(s))
+  if (words.length === 1 && name.length < 10) {
+    return name
+  } else if (words.length < 3) {
+    return name.slice(0,3)
+  } else {
+    return words.map(readableIdPartFromWord).join('')
+  }
+}
+
+class NewFactRow extends Component {
+  state = {
+    variableNameManuallySet: false,
+    fact: {
+      name: '',
+      variable_name: '',
+      expression: '',
+      simulation: {
+        sample: {
+          values: [],
+          errors: [],
+        },
+        stats: {
+          adjustedConfidenceInterval: [],
+          mean: null,
+          stdev: null,
+          length: 0,
+        },
+      },
+    },
+  }
+
+  setFactState(newFactState, otherState = {}) { this.setState({...otherState, fact: {...this.state.fact, ...newFactState}}) }
+  onChangeName(e) {
+    const name = _.get(e, 'target.value')
+    this.setFactState(this.state.variableNameManuallySet ? {name} : {name, variable_name: getVariableNameFromName(name)})
+  }
+  onChangeVariableName(e) { this.setFactState({variable_name: _.get(e, 'target.value')}, {variableNameManuallySet: true}) }
+  onChangeExpression(e) { this.setFactState({expression: _.get(e, 'target.value')}) }
+
+  onBlurExpression() {
+    simulateFact(['biz'], this.state.fact).then(({values, errors}) => {
+      let simulation = {sample: {values, errors}}
+      addStats(simulation)
+      simulation.sample.sortedValues = null // We don't want to send this to the server for latency reasons.
```

Comment:
I realize we don't send this to the server, but can that be decided in the code that sends it to the server, rather than here?